server/src/test/java/org/opensearch/cluster/routing/allocation/BalanceConfigurationTests.java

```diff
@@ -380,6 +413,32 @@ public void testGlobalPrimaryBalance() throws Exception {
         verifyPrimaryBalance(clusterState);
     }
 
+    /**
+     * This test verifies global balance by creating indices iteratively and verify primary shards do not pile up on one
+     * @throws Exception generic exception
+     */
+    public void testGlobalPrimaryBalanceWithNodeDrops() throws Exception {
+        final float buffer = 0.05f;
+        AllocationService strategy = createAllocationService(getSettingsBuilderForPrimaryReBalance().build(), new TestGatewayAllocator());
+        ClusterState clusterState = ClusterState.builder(CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY)).build();
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
+
+        clusterState = addIndex(clusterState, strategy, "test-index1", 5, 1);
+        clusterState = addIndex(clusterState, strategy, "test-index2", 5, 1);
+        clusterState = addIndex(clusterState, strategy, "test-index3", 5, 1);
+
+        clusterState = removeOneNode(clusterState, strategy);
```

Comment:
lets verify primary balance before removing the node as well.

Reply:
Done.